### ISSUE TITLE
Implementation details and hyper-parameter settings

### ISSUE THREAD

**Issue by HowardZJU** [Woeee/FMLP-Rec#3]
Hi! Thanks for such solid work. I have some kind questions about details in implementation.
 1. There are many self-implemented functions, e.g., `LayerNorm`, `gelu`. Why did not use the integrated component in torch? (I have tested the `F.layernorm` would give the same results)
 2. In full-sort setting, is the `seed` set to 42 or else? What about other hyperparameter settings for other models? 
 3. In `FMLPRecDataset`, the last three items are used for training target, eval label and test label. What if the total sequence is no longer than 3? For example, let `self.user_seq = [[1,2,3]]`, the parsed input_ids would be `[[]]` and further padded as 50 zeros, and the trainer seeks to utilize `[[0,0,...,0]]` to forecast `[[1]]`. Maybe exploiting such all-zero sequences to estimate targets would introduce extra noise?

**Comment by Woeee** [comment]
Thanks for your attention and questions.
  1.We utilized self-implemented functions to make codes easier to understand. Obviously, they can be replaced by functions in pytorch.
  2.In full-sort setting, we just set `full_sort` as True, without changing other hyperparameters.
  3.The datasets we used were processed in advance, and the minimum length of sequences is 5. So the situation, what you mentioned like utilize `[0,0,...,0]` to predict `[1]`, will not happen.
I hope my answers can solve your problems.

**Comment by HowardZJU** [comment]
Thanks for your speedy reply~Most of my questions have been addressed.

As for the second question, my understanding is that the hyperparameters (e.g., batchsize, layernum, hidden_dim) are fixed for SASRec and adjusted for FMLP, is it right? Since if I want to reproduce the full-sort results of FMLP, I have to enlarge the batchsize and layer_num.

**Comment by Woeee** [comment]
Yeah, your understanding is correct, and you can find the exact hyperparameters setting in issue #2

**Comment by HowardZJU** [comment]
> Yeah, your understanding is correct, and you can find the exact hyperparameters setting in issue #2

Many thanks for this speedy reply~
Best regards.
